### PR TITLE
Added full API documentation of the  module.

### PR DESCRIPTION
**Pull request by ThomasSteinbach**
What does this PR change? What problem does it solve?
-----------------------------------------------------
* Added full API documentation of the `gcip.core.include` module.

Checklist
---------

- [x] I have read the [Contribution Guidelines](https://github.com/dbsystel/gitlab-ci-python-library/blob/main/CONTRIBUTING.md)
- [x] I have enabled [maintainer edits for this PR](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
- [x] I have added tests for all changes in this PR
- [x] I have added api documentation for the changes
- [x] I added a changelog entry to the unreleased section with the PR number. [Changelog](https://github.com/dbsystel/gitlab-ci-python-library/blob/main/CHANGELOG.md)
- [x] Linting is done, and linting jobs are successfull
- [x] I'm done, this Pull Request is ready for review

### CODE REVIEW

File: gcip/core/include.py [hunk 1]
```diff
@@ -13,19 +65,52 @@
...
+    * `IncludeLocal`
+    * `IncludeFile`
+    * `IncludeRemote`
+    * `IncludeTemplate`
+    * `IncludeArtifact`
+    """
 
     _rendered_include: Dict[str, str]
 
     def render(self) -> Dict[str, str]:
```
Comment (dvonessen) [inline gcip/core/include.py:82]:
```suggestion
    def render(self) -> Dict[str, Any]:
```

File: gcip/core/include.py [hunk 2]
```diff
@@ -13,19 +65,52 @@
...
+            Dict[str, Any]: A dictionary representing the include object in Gitlab CI.
+        """
         return self._rendered_include
 
 
 class IncludeLocal(Include):
+    """This module represents the Gitlab CI [include:local](https://docs.gitlab.com/ee/ci/yaml/#includelocal) keyword.
+
+    Args:
+        local (str): The path to the file within this repository to include.
```
Comment (dvonessen) [inline gcip/core/include.py:98]:
```suggestion
        local (str): Relative path to the file within this repository to include.
```

File: gcip/core/include.py [hunk 3]
```diff
@@ -38,6 +123,15 @@ def __init__(
 
 
 class IncludeRemote(Include):
+    """This module represents the Gitlab CI [include:remote](https://docs.gitlab.com/ee/ci/yaml/#includeremote) keyword.
+
+    Args:
+        remote (str): The URL to include the a file from.
```
Comment (dvonessen) [inline gcip/core/include.py:129]:
```suggestion
        remote (str): URL to include the a file from.
```

File: gcip/core/include.py [hunk 4]
```diff
@@ -13,19 +65,52 @@
...
+
     def __init__(self, local: str) -> None:
         self._rendered_include = {"local": local}
 
 
 class IncludeFile(Include):
+    """This module represents the Gitlab CI [include:file](https://docs.gitlab.com/ee/ci/yaml/#includefile) keyword.
+
+    Args:
+        file (str): The path to the file to include.
```
Comment (dvonessen) [inline gcip/core/include.py:109]:
```suggestion
        file (str): Relative file path to include.
```

File: gcip/core/include.py [hunk 5]
```diff
@@ -13,19 +65,52 @@
...
+        project (str): The project to include the file from.
```
Comment (dvonessen) [inline gcip/core/include.py:110]:
```suggestion
        project (str): Project to include the file from.
```

File: gcip/core/include.py [hunk 6]
```diff
@@ -13,19 +65,52 @@
...
+        ref (Optional[str], optional): The branch of the project to include the file from. Defaults to None.
```
Comment (dvonessen) [inline gcip/core/include.py:111]:
```suggestion
        ref (Optional[str], optional): Projects branch to include the file from. Defaults to None.
```

File: gcip/core/include.py [hunk 7]
```diff
@@ -38,6 +123,15 @@ def __init__(
 
 
 class IncludeRemote(Include):
+    """This module represents the Gitlab CI [include:remote](https://docs.gitlab.com/ee/ci/yaml/#includeremote) keyword.
+
+    Args:
+        remote (str): The URL to include the a file from.
+
+    Raises:
+        ValueError: When the `remote` is not a valid URL.
```
Comment (dvonessen) [inline gcip/core/include.py:132]:
```suggestion
        ValueError: If `remote` is not a valid URL.
```

File: gcip/core/include.py [hunk 8]
```diff
@@ -46,12 +140,24 @@ def __init__(self, remote: str) -> None:
 
 
 class IncludeTemplate(Include):
+    """This module represents the Gitlab CI [include:template](https://docs.gitlab.com/ee/ci/yaml/#includetemplate) keyword.
```
Comment (dvonessen) [inline gcip/core/include.py:143]:
```suggestion
    """This class represents the Gitlab CI [include:template](https://docs.gitlab.com/ee/ci/yaml/#includetemplate) keyword.
```

File: gcip/core/include.py [hunk 9]
```diff
@@ -46,12 +140,24 @@ def __init__(self, remote: str) -> None:
...
+
+    Args:
+        template (str): The Gitlab template pipeline to include.
```
Comment (dvonessen) [inline gcip/core/include.py:146]:
```suggestion
        template (str): Gitlab template pipeline to include.
```

File: gcip/core/include.py [hunk 10]
```diff
@@ -46,12 +140,24 @@ def __init__(self, remote: str) -> None:
...
-    """for triggering a child pipeline with generated configuration file"""
+    """A special type of include: Use a `gcip.core.job.TriggerJob` with `IncludeArtifact` to run [a child pipeline with a generated configuration
+    file from a previous job](https://docs.gitlab.com/ee/ci/yaml/README.html#trigger-child-pipeline-with-generated-configuration-file):
+
+    Args:
+        job (str): The job name to include the artifact from.
+        artifact (str): The name of the job artifact to include.
+    """
 
     def __init__(self, job: str, artifact: str):
```
Comment (dvonessen) [inline gcip/core/include.py:162]:
```suggestion
    def __init__(self, job: str, artifact: str) -> None:
```

File: gcip/core/include.py [hunk 11]
```diff
@@ -46,12 +140,24 @@ def __init__(self, remote: str) -> None:
...
         self._rendered_include = {"template": template}
 
 
 class IncludeArtifact(Include):
-    """for triggering a child pipeline with generated configuration file"""
+    """A special type of include: Use a `gcip.core.job.TriggerJob` with `IncludeArtifact` to run [a child pipeline with a generated configuration
+    file from a previous job](https://docs.gitlab.com/ee/ci/yaml/README.html#trigger-child-pipeline-with-generated-configuration-file):
+
+    Args:
+        job (str): The job name to include the artifact from.
```
Comment (dvonessen) [inline gcip/core/include.py:158]:
```suggestion
        job (str): Job name to include the artifact from.
```

File: gcip/core/include.py [hunk 12]
```diff
@@ -46,12 +140,24 @@ def __init__(self, remote: str) -> None:
...
+        artifact (str): The name of the job artifact to include.
```
Comment (dvonessen) [inline gcip/core/include.py:159]:
```suggestion
        artifact (str): Relative path to artifact which is produced by `job`.
```

File: gcip/core/include.py (hunk 1 above)
Comment (ThomasSteinbach) [reply gcip/core/include.py:82]: `_rendered_include` is of `Dict[str, str]`. But you're right - I provided the wrong type in the documentation. I'll fix that.

File: gcip/core/include.py (hunk 3 above)
Comment (ThomasSteinbach) [reply gcip/core/include.py:129]:
```suggestion
        remote (str): URL to include the file from.
```

File: gcip/core/include.py (hunk 4 above)
Comment (ThomasSteinbach) [reply gcip/core/include.py:109]:
```suggestion
        file (str): Relative path to the file to include.
```

File: gcip/core/include.py (hunk 6 above)
Comment (ThomasSteinbach) [reply gcip/core/include.py:111]:
```suggestion
        ref (Optional[str], optional): Project branch to include the file from. Defaults to None.
```

File: gcip/core/include.py (hunk 12 above)
Comment (ThomasSteinbach) [reply gcip/core/include.py:159]:
```suggestion
        artifact (str): Relative path to the artifact which is produced by `job`.
```

**Review by dvonessen — APPROVED** [review]

**Pull request merged**